apps/explorer/app/contract/[id]/page.tsx

```diff
@@ -24,116 +24,115 @@ export function generateMetadata({ params }): Metadata {
 export const revalidate = 0
 
 export default async function Page({ params }) {
-  const id = params?.id as string
-  const [[c, error], [r]] = await Promise.all([
-    to(
-      siaCentral.contract({
-        params: {
-          id,
-        },
-      })
-    ),
+  const id = params?.id
+
+  // Grab the contract and previous revisions data.
+  const [
+    [rate, rateError],
+    [contract, contractError],
+    [previousRevisions, previousRevisionsError],
+    [currentTip, currentTipError],
+  ] = await Promise.all([
     to(
       siaCentral.exchangeRates({
         params: {
           currencies: 'sc',
         },
       })
     ),
+    to(explored.contractByID({ params: { id } })),
+    to<ExplorerFileContract[]>(explored.contractRevisions({ params: { id } })),
+    to(explored.consensusTip()),
   ])
 
-  if (error) {
-    throw error
-  }
+  if (rateError) throw rateError
+  if (!rate) throw 'No rate found in successful request'
+
+  if (contractError) throw contractError
+  if (!contract) return notFound()
 
-  const contract = c?.contract
+  if (previousRevisionsError) throw previousRevisionsError
+  if (!previousRevisions)
+    throw 'No previousRevisions found in successful request'
 
-  if (!contract) {
-    return notFound()
-  }
+  if (currentTipError) throw currentTipError
+  if (!currentTip) throw 'No currentTip found in successful request'
```

Comment:
if (!rate) throw new Error('No currentTip found in successful request')

Reply:
What does this mean?